marketplacetests/payment/app.py

```diff
@@ -140,4 +143,7 @@ def _tap_payment_button(self, button_locator):
         self.marionette.find_element(*button_locator).tap()
         self.marionette.switch_to_frame()
         self.wait_for_element_not_present(*self._payment_frame_locator)
-        self.switch_to_marketplace_frame()
+        if return_to == 'marketplace':
+            self.switch_to_marketplace_frame()
+        else:
+            self.apps.switch_to_displayed_app()
```

Comment:
I'm not crazy about this logic. Could we have an InAppPayment that extends Payment and overrides this method (and any others, as needed)?